src/Components/Aspire.Seq/AspireSeqExtensions.cs

```diff
@@ -37,39 +45,45 @@ public static void AddSeqEndpoint(this IHostApplicationBuilder builder, string c
 
         configureSettings?.Invoke(settings);
 
-        if (string.IsNullOrEmpty(settings.ServerUrl))
+        if (!string.IsNullOrEmpty(settings.ServerUrl))
         {
-            settings.ServerUrl = "http://localhost:5341";
+            settings.Logs.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/logs");
+            settings.Traces.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/traces");
         }
-
-        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddOtlpExporter(opt =>
+        if (!string.IsNullOrEmpty(settings.ApiKey))
         {
-            opt.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/logs");
-            opt.Protocol = OtlpExportProtocol.HttpProtobuf;
-            if (!string.IsNullOrEmpty(settings.ApiKey))
-            {
-                opt.Headers = $"X-Seq-ApiKey={settings.ApiKey}";
-            }
-        }));
-        builder.Services.ConfigureOpenTelemetryTracerProvider(tracing => tracing
-            .AddOtlpExporter(opt =>
-                {
-                    opt.Endpoint = new Uri($"{settings.ServerUrl}/ingest/otlp/v1/traces");
-                    opt.Protocol = OtlpExportProtocol.HttpProtobuf;
-                    if (!string.IsNullOrEmpty(settings.ApiKey))
-                    {
-                        opt.Headers = $"X-Seq-ApiKey={settings.ApiKey}";
-                    }
-                }
+            settings.Logs.Headers = string.IsNullOrEmpty(settings.Logs.Headers) ? $"X-Seq-ApiKey={settings.ApiKey}" : $"{settings.Logs.Headers},X-Seq-ApiKey={settings.ApiKey}";
+            settings.Traces.Headers = string.IsNullOrEmpty(settings.Traces.Headers) ? $"X-Seq-ApiKey={settings.ApiKey}" : $"{settings.Traces.Headers},X-Seq-ApiKey={settings.ApiKey}";
+        }
+
+        builder.Services.Configure<OpenTelemetryLoggerOptions>(logging => logging.AddProcessor(
+            _ => settings.Logs.ExportProcessorType == ExportProcessorType.Batch
+                ? new BatchLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
+                : new SimpleLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
```

Comment:
switch expressions are much cleaner here:

```suggestion
            _ => settings.Logs.ExportProcessorType switch
                ExportProcessorType.Batch => new BatchLogRecordExportProcessor(new OtlpLogExporter(settings.Logs)),
                _ => new SimpleLogRecordExportProcessor(new OtlpLogExporter(settings.Logs))
```

Reply:
Thanks @davidfowl 

This change is done.